srml/support/src/storage/hashed/generator.rs

```diff
@@ -203,23 +208,64 @@ pub trait StorageValue<T: codec::Codec> {
 	///
 	/// `T` is required to implement `codec::EncodeAppend`.
 	fn append<S: HashedStorage<Twox128>, I: codec::Encode>(
-		items: &[I], storage: &mut S
-	) -> Result<(), &'static str> where T: codec::EncodeAppend<Item=I> {
+		items: &[I],
+		storage: &mut S,
+	) -> Result<(), &'static str>
+		where T: codec::EncodeAppend<Item=I>
+	{
 		let new_val = <T as codec::EncodeAppend>::append(
-			storage.get_raw(Self::key()).unwrap_or_default(),
+			// if the key exists, directly append to it.
+			storage.get_raw(Self::key()).unwrap_or_else(|| {
+				// otherwise, try and read a proper __provided__ default.
+				Self::Default::default().map(|v| v.encode())
+					// or just use the default value.
```

Comment:
This comment is misleading. In the context you first think that you talk about the `Self::Default` stuff. Actually we just use `Vec::default()`.

Reply:
the goal was to de-mistify this by explicitly mentioning in the line above: `**provided** default value` so that once knows by `default value` I mean rust's `Default`. made explicit now 👍